@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -928,12 +952,11 @@ export class TextElementsRenderer {
 
         this.m_cacheInvalidated = false;
 
-        const renderList = this.m_mapView.visibleTileSet.dataSourceTileList;
-
-        this.checkIfOverloaded();
+        this.checkIfOverloaded(dataSourceTileList);
 
+        const worldMaxX = projection.worldExtent(0, 0).max.x;
```

Comment:
I understand the need to move this out for performance reasons, however moving it has made it harder to track what this is used for, because you have decoupled the offset from this value. Please add a comment to say this is used in conjunction with the tile offset.

Reply:
done